ls1intum/Artemis
# Fix non working authentication over LTI

This should fix an issue with the authentication over LTI
We currently assume that the forwarding of the full URL is not working correctly and therefore the request cannot be verified.
It seems that Spring does not check the header value 'X-Forwarded-Proto' set by nginx.

### Motivation and Context
Users of online courses cannot participate in exercises over LTI

### Description
Additionally, I made the whole LTI config optional, so that administrators of other Artemis instances can also deactivate it.

### Steps for Testing
0. Make sure you are logged out on Artemis
1. Create an account on edx (do not use your TUM Email) and register for the software engineering essentials course
2. Participate in one programming exercises (use OOP1 because this is synced with TS1) over LTI
3. You should automatically get a new Artemis account (starting with 'edx_') when you first try this. There should be no issue with unauthorized errors
